--- patch-2.6.25-rc8-git7	2008-04-08 22:17:01.000000000 -0600
+++ patch-libre-2.6.25-rc8-git7	2008-04-11 11:21:34.000000000 -0600
@@ -1,3 +1,7 @@
+Deblobbed
+
+drivers/media/Makefile - refers to disabled/removed bits.
+
 diff --git a/Documentation/SubmittingPatches b/Documentation/SubmittingPatches
 index 08a1ed1..47a539c 100644
 --- a/Documentation/SubmittingPatches
@@ -18387,7 +18391,7 @@
 index 8cf9135..7b8bb69 100644
 --- a/drivers/media/Makefile
 +++ b/drivers/media/Makefile
-@@ -6,3 +6,6 @@ obj-y := common/
+# @@ -6,3 +6,6 @@ obj-y := common/
  obj-y += video/
  obj-$(CONFIG_VIDEO_DEV) += radio/
  obj-$(CONFIG_DVB_CORE)  += dvb/
