Index: debian/ruleset/misc/version_vars.mk
===================================================================
--- debian/ruleset/misc/version_vars.mk.orig	2008-11-24 17:01:32.000000000 +0000
+++ debian/ruleset/misc/version_vars.mk	2010-12-12 13:57:33.000000000 +0000
@@ -138,7 +138,9 @@
 EXTRAV_ARG :=
 endif
 
-UTS_RELEASE_HEADER=$(call doit,if [ -f include/linux/utsrelease.h ]; then  \
+UTS_RELEASE_HEADER=$(call doit,if [ -f include/generated/utsrelease.h ]; then  \
+	                       echo include/generated/utsrelease.h;            \
+	                   elif [ -f include/linux/utsrelease.h ]; then  \
 	                       echo include/linux/utsrelease.h;            \
 	                   else                                            \
                                echo include/linux/version.h ;              \
Index: debian/ruleset/targets/common.mk
===================================================================
--- debian/ruleset/targets/common.mk.orig	2008-11-24 17:01:32.000000000 +0000
+++ debian/ruleset/targets/common.mk	2010-12-12 13:57:33.000000000 +0000
@@ -333,7 +333,7 @@
 	@echo this was built on a machine with the kernel: >> debian/buildinfo
 	uname -a >> debian/buildinfo
 	echo using the compiler: >> debian/buildinfo
-	grep LINUX_COMPILER include/linux/compile.h | \
+	grep LINUX_COMPILER include/generated/compile.h | \
 	   sed -e 's/.*LINUX_COMPILER "//' -e 's/"$$//' >> debian/buildinfo
 ifneq ($(strip $(shell test -f version.Debian && cat version.Debian)),)
 	echo kernel source package used: >> debian/buildinfo
